src/windows/wslaservice/exe/WSLAUserSession.h

```diff
@@ -28,14 +30,18 @@ class WSLAUserSessionImpl
     PSID GetUserSid() const;
 
     HRESULT CreateVirtualMachine(const VIRTUAL_MACHINE_SETTINGS* Settings, IWSLAVirtualMachine** VirtualMachine);
+    HRESULT CreateSession(const WSLA_SESSION_SETTINGS* Settings, IWSLASession** Session);
 
     void OnVmTerminated(WSLAVirtualMachine* machine);
 
 private:
     wil::unique_tokeninfo_ptr<TOKEN_USER> m_tokenInfo;
 
-    std::recursive_mutex m_virtualMachinesLock;
+    std::recursive_mutex m_lock;
     std::vector<WSLAVirtualMachine*> m_virtualMachines;
+
+    // TODO-WSLA: Consider using a weak_ptr to easily destroy when the last client reference is released.
+    std::vector<Microsoft::WRL::ComPtr<WSLASession>> m_sessions;
```

Comment:
>m_sessions [](http://example.com/codeflow?start=53&length=10)

What happens if we don't keep a reference to the com pointer in the service? My understanding was that the lifetime would be controlled by the client?